cmd/oklog/retrieve.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/oklog/ulid"
+	"github.com/pkg/errors"
+	"storj.io/uplink"
+)
+
+func runRetrieve(args []string) error {
+	flagset := flag.NewFlagSet("retrieve", flag.ExitOnError)
+	var (
+		bucketName  = flagset.String("bucket", "", "bucket name to retrieve logs from")
+		accessGrant = flagset.String("access", os.Getenv("ACCESS_GRANT"), "access grant from satellite")
```

Comment:
Wouldn’t this access grant already exist in the context of the application? For pushing?

Reply:
I'm assuming the retrieve command still hits the oklog API endpoint?

Reply:
@kristaxox Yes, absolutely right, I still need to shift the uplink stuff into an API endpoint and have this command just query that :)